test_conformance/basic/test_global_linear_id.cpp

```diff
@@ -22,105 +22,93 @@
-                             NULL, &err);
-    test_error( err, "clCreateBuffer failed.");
-
-    err = create_single_kernel_helper(context, &program[0], &kernel[0], 1,
-                                      &global_linear_id_1d_code,
-                                      "test_global_linear_id_1d");
-    test_error( err, "create_single_kernel_helper failed");
-    err = create_single_kernel_helper(context, &program[1], &kernel[1], 1,
-                                      &global_linear_id_2d_code,
-                                      "test_global_linear_id_2d");
-    test_error( err, "create_single_kernel_helper failed");
-
-    err  = clSetKernelArg(kernel[0], 0, sizeof streams, &streams);
-    test_error( err, "clSetKernelArgs failed.");
-    err  = clSetKernelArg(kernel[1], 0, sizeof streams, &streams);
-    test_error( err, "clSetKernelArgs failed.");
-
-    threads[0] = (size_t)num_elements;
-    threads[1] = (size_t)num_elements;
-    err = clEnqueueNDRangeKernel(queue, kernel[1], 2, NULL, threads, NULL, 0, NULL, NULL);
-    test_error( err, "clEnqueueNDRangeKernel failed.");
-
-    err = clEnqueueReadBuffer(queue, streams, CL_TRUE, 0, length*sizeof(int), output_ptr, 0, NULL, NULL);
-    test_error( err, "clEnqueueReadBuffer failed.");
-
-    err = verify_global_linear_id(output_ptr, length);
-
-    threads[0] = (size_t)num_elements;
-    err = clEnqueueNDRangeKernel(queue, kernel[1], 1, NULL, threads, NULL, 0, NULL, NULL);
-    test_error( err, "clEnqueueNDRangeKernel failed.");
-
-    err = clEnqueueReadBuffer(queue, streams, CL_TRUE, 0, num_elements*sizeof(int), output_ptr, 0, NULL, NULL);
-    test_error( err, "clEnqueueReadBuffer failed.");
-
-    err = verify_global_linear_id(output_ptr, num_elements);
-
-    // cleanup
-    clReleaseMemObject(streams);
-    clReleaseKernel(kernel[0]);
-    clReleaseKernel(kernel[1]);
-    clReleaseProgram(program[0]);
-    clReleaseProgram(program[1]);
-    free(output_ptr);
+                             nullptr, &err);
+    test_error(err, "clCreateBuffer failed.");
+
+    for (int i = 0; i < ARRAY_SIZE(program); i++)
+    {
+        err = create_single_kernel_helper(context, &program[i], &kernel[i], 1,
+                                          &kernel_code[i], kernel_names[i]);
+        test_error(err, "create_single_kernel_helper failed");
+
+        err = clSetKernelArg(kernel[i], 0, sizeof streams, &streams);
+        test_error(err, "clSetKernelArgs failed.");
+
+        err = clEnqueueNDRangeKernel(queue, kernel[i], i + 1, nullptr, threads,
+                                     nullptr, 0, nullptr, nullptr);
+        test_error(err, "clEnqueueNDRangeKernel failed.");
+
+        err = clEnqueueReadBuffer(queue, streams, CL_TRUE, 0,
+                                  length * sizeof(int), output.data(), 0,
+                                  nullptr, nullptr);
+        test_error(err, "clEnqueueReadBuffer failed.");
+
+        err = verify_global_linear_id(output, length);
```

Comment:
Verification probably fails in the first iteration of the loop (see comment above) but the status isn't checked and in the second iteration the error gets overwritten.